go/libraries/doltcore/mvdata/engine_table_writer.go

```diff
@@ -265,52 +265,70 @@ func (s *SqlEngineTableWriter) createOrEmptyTableIfNeeded() error {
 
 // createTable creates a table.
 func (s *SqlEngineTableWriter) createTable() error {
-	cr := plan.NewCreateTable(sql.UnresolvedDatabase(s.database), s.tableName, false, false, &plan.TableSpec{Schema: s.tableSchema})
-	analyzed, err := s.se.Analyze(s.sqlCtx, cr)
-	if err != nil {
-		return err
+	// TODO don't use internal interfaces to do this, we had to have a sql.Schema somewhere
+	// upstream to make the dolt schema
+	sqlCols := make([]string, len(s.tableSchema.Schema))
+	for i, c := range s.tableSchema.Schema {
+		sqlCols[i] = sql.GenerateCreateTableColumnDefinition(c, c.Default.String())
 	}
-
-	analyzedQueryProcess := analyzer.StripPassthroughNodes(analyzed.(*plan.QueryProcess))
-
-	ri, err := rowexec.DefaultBuilder.Build(s.sqlCtx, analyzedQueryProcess, nil)
-	if err != nil {
-		return err
+	var pks string
+	var sep string
+	for _, i := range s.tableSchema.PkOrdinals {
+		pks += sep + s.tableSchema.Schema[i].Name
+		sep = ", "
 	}
-
-	for {
-		_, err = ri.Next(s.sqlCtx)
-		if err != nil {
-			return ri.Close(s.sqlCtx)
-		}
+	if len(sep) > 0 {
+		sqlCols = append(sqlCols, fmt.Sprintf("PRIMARY KEY (%s)", pks))
 	}
-}
 
-// getInsertNode returns the sql.Node to be iterated on given the import option.
-func (s *SqlEngineTableWriter) getInsertNode(inputChannel chan sql.Row) (sql.Node, error) {
-	switch s.importOption {
-	case CreateOp, ReplaceOp, AppendOp:
-		return s.createInsertImportNode(inputChannel, s.contOnErr, false, nil) // contonerr translates to ignore
-	case UpdateOp:
-		return s.createInsertImportNode(inputChannel, s.contOnErr, false, generateOnDuplicateKeyExpressions(s.rowOperationSchema.Schema)) // contonerr translates to ignore
-	default:
-		return nil, fmt.Errorf("unsupported import type")
+	createTable := sql.GenerateCreateTableStatement(s.tableName, sqlCols, sql.CharacterSet_utf8.String(), sql.Collation_Default.String())
+	sch, iter, err := s.se.Query(s.sqlCtx, createTable)
+	if err != nil {
+		return err
 	}
+	_, err = sql.RowIterToRows(s.sqlCtx, sch, iter)
+	return err
 }
 
 // createInsertImportNode creates the relevant/analyzed insert node given the import option. This insert node is wrapped
```

Comment:
method doc out of date

Reply:
I think the doc is still accurate? We are still making an analyzed insert node, it's just using a different interface